File: src/main/java/ac/dnd/server/admission/domain/model/ApplicantNameEmailBlindIndex.java
package ac.dnd.server.admission.domain.model;

import io.micrometer.common.util.StringUtils;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ApplicantNameEmailBlindIndex {
	private String name;
	private String email;

	private ApplicantNameEmailBlindIndex(
		final String name,
		final String email
	) {

		if (StringUtils.isEmpty(name)) {
			throw new IllegalArgumentException("시작일은 null일 수 없습니다.");
		}

		if (StringUtils.isEmpty(email)) {
			throw new IllegalArgumentException("시작일은 null일 수 없습니다.");
		}

		this.email = email;
		this.name = name;
	}

	public static ApplicantNameEmailBlindIndex of(
		final String name,
		final String email
	) {
		return new ApplicantNameEmailBlindIndex(
			name,
			email
		);
	}

}

File: src/main/kotlin/ac/dnd/server/admission/domain/model/ApplicantNameEmailBlindIndex.kt
package ac.dnd.server.admission.domain.model

data class ApplicantNameEmailBlindIndex(
    val nameBlindIndex: String,
    val emailBlindIndex: String
) {
    init {
        require(nameBlindIndex.isNotBlank()) {
            "이름 블라인드 인덱스(nameBlindIndex)는 공백일 수 없습니다."
        }
        require(emailBlindIndex.isNotBlank()) {
            "이메일 블라인드 인덱스(emailBlindIndex)는 공백일 수 없습니다."
        }
    }

    companion object {
        fun of(nameBlindIndex: String, emailBlindIndex: String): ApplicantNameEmailBlindIndex {
            return ApplicantNameEmailBlindIndex(nameBlindIndex, emailBlindIndex)
        }
    }
}